internal/api/auth.go

```diff
@@ -35,6 +35,15 @@ func (a *API) requireAuthentication(w http.ResponseWriter, r *http.Request) (con
 	return ctx, err
 }
 
+func (a *API) requireNotAnonymous(w http.ResponseWriter, r *http.Request) (context.Context, error) {
+	ctx := r.Context()
+	claims := getClaims(ctx)
+	if claims.IsAnonymous {
+		return nil, unauthorizedError("Anonymous user not allowed")
```

Comment:
This should be:

```suggestion
		return nil, unauthorizedError("Anonymous user not allowed to perform these actions")
```

Unauthorized means the `Authorization` header is missing. Forbidden means there's insufficient permissions.